upup/pkg/fi/nodeup/command.go

```diff
@@ -462,6 +462,25 @@ func evaluateHostnameOverride(hostnameOverride string) (string, error) {
 		return domain, nil
 	}
 
+	if k == "@aws-private" {
```

Comment:
If we go with this approach, I'd also like to swap this section to also switch on the `k` -- not sure why its a lot of if blocks now :/